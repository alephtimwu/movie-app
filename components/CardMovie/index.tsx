import React, { useEffect, useState } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Rating } from 'react-native-elements';
import globalStyles from '../../constants/styles';
import {
  StyledCard,
  StyledPoster,
  StyledTitleWrapper,
  StyledTitle,
  StyledReviewWrapper,
  StyledRating,
  StyledReviews
} from './style';

interface MovieItem {
  Poster: string;
  Title: string;
  Year: string;
  imdbID: string;
  Type: string;
}

interface CardMovieProps {
  movie: MovieItem;
  onPress?: (id) => void;
}

export default ({ movie, onPress }: CardMovieProps) => {
  const { Poster, Title, Year, imdbID } = movie;

  const handlePressMovie = () => {
    if (onPress) {
      onPress(imdbID);
    }
  };

  return (
    <StyledCard>
      <TouchableOpacity onPress={handlePressMovie}>
        <StyledPoster source={{ uri: Poster }} />
      </TouchableOpacity>
      <StyledTitleWrapper>
        <StyledTitle numberOfLines={1}>{Title}</StyledTitle>
      </StyledTitleWrapper>
      <StyledReviewWrapper>
        {/* <StyledRating>
          <Rating
            type="custom"
            showRating={false}
            readonly={true}
            ratingCount={5}
            imageSize={8}
          />
        </StyledRating> */}
        <StyledReviews>{Year}</StyledReviews>
        <StyledReviews>{` - ${imdbID}`}</StyledReviews>
      </StyledReviewWrapper>
    </StyledCard>
  );
};
