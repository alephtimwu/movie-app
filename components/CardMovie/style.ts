import styled from '@emotion/native';

export const StyledCard = styled.View`
  margin-right: 20px;
`;
export const StyledPoster = styled.Image`
  ${({ width }) => `width: ${width || 230}px;`}
  ${({ height }) => `height: ${height || 340}px;`}
`;

export const StyledTitleWrapper = styled.View`
  margin-top: 10px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  ${({ width }) => `width: ${width ? width - 20 : 210}px;`}
`;
export const StyledTitle = styled.Text`
  text-align: left;
  font-family: Montserrat;
  color: #fff;
  font-size: 16px;
  line-height: 16px;
`;

export const StyledReviewWrapper = styled.View`
  margin-top: 10px;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
`;

export const StyledRating = styled.View`
  background-color: #1d1d27;
  margin-right: 5px;
`;

export const StyledReviews = styled.Text`
  text-align: left;
  font-family: Montserrat;
  color: #fff;
  font-size: 10px;
  line-height: 10px;
`;
