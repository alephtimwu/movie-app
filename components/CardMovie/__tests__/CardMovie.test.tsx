import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
// import { CrpAll } from '__mocks__';
import mockMovie from '../../../constants/movies';
import CardMovie from '../index';

describe('CardMovie =>', () => {
  let tree = null;
  const handlePress = jest.fn();

  beforeEach(() => {
    tree = render(
      <CardMovie movie={mockMovie.mock[0]} onPress={handlePress} />
    );
  });

  test('CardMovie should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('CardMovie should render correctly with custom width height', () => {
    const { rerender, getByTestId } = tree;
    rerender(
      <CardMovie
        movie={mockMovie.mock[0]}
        onPress={handlePress}
        width={80}
        height={100}
      />
    );
    fireEvent.press(getByTestId('card-press'));
    expect(handlePress).toBeCalled();
  });
});
