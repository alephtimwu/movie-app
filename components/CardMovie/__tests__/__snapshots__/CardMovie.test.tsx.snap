// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMovie => CardMovie should render correctly 1`] = `
<View
  style={
    Object {
      "marginRight": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="card-press"
  >
    <Image
      source={
        Object {
          "uri": "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg",
        }
      }
      style={
        Object {
          "height": 340,
          "width": 230,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
        "overflow": "hidden",
        "width": 210,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Montserrat",
          "fontSize": 16,
          "lineHeight": 16,
          "textAlign": "left",
        }
      }
    >
      Captain Marvel Captain Marvel Captain Marvel
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Montserrat",
          "fontSize": 10,
          "lineHeight": 10,
          "textAlign": "left",
        }
      }
    >
      2019
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Montserrat",
          "fontSize": 10,
          "lineHeight": 10,
          "textAlign": "left",
        }
      }
    >
       - tt4154664
    </Text>
  </View>
</View>
`;
